### PR TITLE
feat(manager): add sveltos manager

### PR DESCRIPTION
**Pull request by oliverbaehler**
## Changes

This adds the manager for [sveltos](https://projectsveltos.github.io/sveltos/) for GitOps managed Cluster Fleets. Essentially we have the same functionality as argoCD, so we were able to take the argocd manager and adjust the schema etc. I hope that's okay.

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

Not sure if there's more?

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by secustor** [comment]
Please create first a discussion with the 'new package manager' label and the questionnaire filled out, so we have all information to make decision. 

I'm not closing this, but moving it into draft till then as it looks well crafted on a first glance.

**Comment by oliverbaehler** [comment]
@secustor thanks for your feedback, i hope i have correctly created the discussion: https://github.com/renovatebot/renovate/discussions/30090

However i cant add the mentioned label, do you need to add it?

**Comment by secustor** [comment]
Package manager details can be found here: https://github.com/renovatebot/renovate/discussions/30090#discussioncomment-9990664

File: lib/modules/manager/sveltos/extract.spec.ts
Comment (secustor) [inline lib/modules/manager/sveltos/extract.spec.ts:37]:
```suggestion
        codeblock`
```
Use `codeblock` to allow indentation of the inline fixtures. 
https://github.com/renovatebot/renovate/blob/057049b134f7987e5029f435f3d2a797bdccd04f/lib/util/toml.spec.ts#L6

File: lib/modules/manager/sveltos/__fixtures__/validEventTrigger.yml
Comment (secustor) [inline lib/modules/manager/sveltos/__fixtures__/validEventTrigger.yml:18]: Could people want to create rules specific to `EventTrigger`, `ClusterProfile` or `Profile`?

File: lib/modules/manager/sveltos/index.ts
Comment (secustor) [inline lib/modules/manager/sveltos/index.ts:19]:
```suggestion
```
I don't see any usages of this currently.

File: lib/modules/manager/sveltos/extract.ts
Comment (secustor) [inline lib/modules/manager/sveltos/extract.ts:74]:
```suggestion
  definition: ProfileDefinition,
```

File: lib/modules/manager/sveltos/extract.ts
Comment (secustor) [inline lib/modules/manager/sveltos/extract.ts:84]: Where is this sources coming from?
I think this whole function needs a rework.

**Review by secustor — CHANGES_REQUESTED** [review]

File: lib/modules/manager/sveltos/extract.ts
Comment (oliverbaehler) [reply lib/modules/manager/sveltos/extract.ts:84]: Copied from ArgoCD https://github.com/renovatebot/renovate/blob/d65effaa77b53ef93312474e81fdbd0efeff0498/lib/modules/manager/argocd/extract.ts#L81

Do you have any hints what needs to be changed (I am a ts noob) :D

File: lib/modules/manager/sveltos/__fixtures__/validEventTrigger.yml
Comment (oliverbaehler) [reply lib/modules/manager/sveltos/__fixtures__/validEventTrigger.yml:18]: It's probably more common to have them all combined instead of making specific rules. Should i combine the fixtures?

Comment (secustor) [reply lib/modules/manager/sveltos/__fixtures__/validEventTrigger.yml:18]: I'm referring here to `depType`s. See this section of the `terraform` manager as an example. https://docs.renovatebot.com/modules/manager/terraform/#disabling-parts-of-the-manager

File: lib/modules/manager/sveltos/extract.ts
Comment (secustor) [reply lib/modules/manager/sveltos/extract.ts:84]: You are trying to access a field which does not exist on the object. 
`spec.source` and `spec.sources` references these fields: 
- https://github.com/renovatebot/renovate/blob/d65effaa77b53ef93312474e81fdbd0efeff0498/lib/modules/manager/argocd/__fixtures__/validApplication.yml#L66
- https://github.com/renovatebot/renovate/blob/d65effaa77b53ef93312474e81fdbd0efeff0498/lib/modules/manager/argocd/__fixtures__/validApplication.yml#L82

**Review by secustor — CHANGES_REQUESTED** [review]
Also needs prettier run. Check the contributor docs https://docs.renovatebot.com/contributing-to-renovate/

**Comment by oliverbaehler** [comment]
@secustor Implemented suggestions

**Comment by secustor** [comment]
Needs fixes and do not use force push as we squash the commits anyway while merging. 
Further please use the rereview function instead of tagging.

**Comment by oliverbaehler** [comment]
Sorry, forced pushed again. Won't do again!

**Comment by oliverbaehler** [comment]
@secustor Yes this is ready for review, Sorry!

File: lib/modules/manager/sveltos/readme.md
Comment (HonkingGoose) [inline lib/modules/manager/sveltos/readme.md:3]:
```suggestion
### You must set a `fileMatch` pattern
```

Put the key info in the heading: the reader must do something.

File: lib/modules/manager/sveltos/readme.md
Comment (HonkingGoose) [inline lib/modules/manager/sveltos/readme.md:6]:
```suggestion
#### fileMatch pattern examples

```

File: lib/modules/manager/sveltos/readme.md
Comment (HonkingGoose) [inline lib/modules/manager/sveltos/readme.md:31]:
```suggestion
### Disabling parts of the sveltos manager
```

File: lib/modules/manager/sveltos/readme.md
Comment (HonkingGoose) [inline lib/modules/manager/sveltos/readme.md:36]:
```suggestion
| :-------------------------------------------------------------------------------------- | :--------------- |
```

Align all text to the left.

**Review by HonkingGoose — CHANGES_REQUESTED** [review]

**Review by secustor — CHANGES_REQUESTED** [review]
Needs code coverage ( see the failing CI ) and docs changes requested by `@HonkingGoose`

File: lib/modules/manager/sveltos/readme.md
Comment (viceice) [reply lib/modules/manager/sveltos/readme.md:3]: we've other managers with empty file match, we should align them to use same layout

File: lib/modules/manager/sveltos/schema.ts
Comment (oliverbaehler) [reply lib/modules/manager/sveltos/schema.ts:4]: No, i have extended the object. Is that okay or not?

File: lib/modules/manager/sveltos/extract.spec.ts
Comment (viceice) [inline lib/modules/manager/sveltos/extract.spec.ts:8]: move to test if only used once (perf)

File: lib/modules/manager/sveltos/extract.ts
Comment (viceice) [inline lib/modules/manager/sveltos/extract.ts:24]:
```suggestion
    definitions = parseYaml(content, {
```

File: lib/modules/manager/sveltos/schema.ts
Comment (viceice) [reply lib/modules/manager/sveltos/schema.ts:4]: we should unify it

https://github.com/renovatebot/renovate/blob/302718f38e54c6c7590f488d749fb4da8362f36f/lib/modules/manager/flux/schema.ts#L3-L13

File: lib/modules/manager/sveltos/extract.ts
Comment (viceice) [inline lib/modules/manager/sveltos/extract.ts:20]: we should support `config.registryAliases`.

https://github.com/renovatebot/renovate/blob/302718f38e54c6c7590f488d749fb4da8362f36f/lib/modules/manager/flux/extract.ts#L150-L177

File: lib/modules/manager/sveltos/extract.ts
Comment (secustor) [inline lib/modules/manager/sveltos/extract.ts:26]:
```suggestion
    });
```
Do not use any

File: lib/modules/manager/sveltos/extract.ts
Comment (viceice) [inline lib/modules/manager/sveltos/extract.ts:22]:
```suggestion
  let definitions: ProfileDefinition[];
```

File: lib/modules/manager/sveltos/extract.ts
Comment (viceice) [inline lib/modules/manager/sveltos/extract.ts:39]: extract util function

Comment (oliverbaehler) [reply lib/modules/manager/sveltos/extract.ts]: refactored the function (i hope that was you were talking about)

**Review by viceice — DISMISSED** [review]

File: lib/modules/manager/sveltos/extract.ts
Comment (secustor) [inline lib/modules/manager/sveltos/extract.ts:25]:
```suggestion
    definitions = parseYaml(content, {
```
This still need to be updated

**Review by secustor — COMMENTED** [review]
You are missing coverage.

**Comment by oliverbaehler** [comment]
Hi I am sorry I am unable to get the coverage up :( . Can I somehow create the coverage locally? What i currently do is run Jest:

```
pnpm jest lib/modules/manager/sveltos
...
Statements   : 2.39% ( 1023/42686 )
Branches     : 0.36% ( 49/13404 )
Functions    : 0.51% ( 27/5294 )
Lines        : 2.42% ( 1020/42083 )
```

But i don't know how to interpret these results..

**Comment by rarkins** [comment]
https://github.com/renovatebot/renovate/blob/9b64288861fcebfcce7a4938307dec52c04ec54f/docs/development/local-development.md?plain=1#L157-L166

File: lib/modules/manager/sveltos/extract.spec.ts
Comment (viceice) [inline lib/modules/manager/sveltos/extract.spec.ts:4]:
```suggestion
import type { ProfileDefinition } from './schema';
```

File: lib/modules/manager/sveltos/extract.spec.ts
Comment (viceice) [inline lib/modules/manager/sveltos/extract.spec.ts:31]: wrong, doesn't work this way

File: lib/modules/manager/sveltos/extract.ts
Comment (viceice) [inline lib/modules/manager/sveltos/extract.ts:44]: why parsing again? makes no sense as it's already validated in caller. remove and inline this function.

File: lib/modules/manager/sveltos/extract.ts
Comment (viceice) [inline lib/modules/manager/sveltos/extract.ts:35]: use for-of loop

Comment (oliverbaehler) [reply lib/modules/manager/sveltos/extract.ts]: adjusted, recheck

File: .github/workflows/undesirable-test-additions.yaml
Comment (rarkins) [inline .github/workflows/undesirable-test-additions.yaml:18]:
```suggestion
```

File: lib/modules/manager/sveltos/extract.spec.ts
Comment (HonkingGoose) [inline lib/modules/manager/sveltos/extract.spec.ts:56]:
```suggestion
    it('return null if YAML is invalid', () => {
```

File: lib/modules/manager/sveltos/readme.md
Comment (HonkingGoose) [inline lib/modules/manager/sveltos/readme.md:11]:
```suggestion
#### `fileMatch` pattern examples
```

Let's use the `code` styling for the config option name here too.

**Review by HonkingGoose — CHANGES_REQUESTED** [review]

File: lib/modules/manager/sveltos/extract.ts
Comment (secustor) [inline lib/modules/manager/sveltos/extract.ts:98]:
```suggestion
    for (const source of coerceArray(definition.spec?.helmCharts)) {
      const dep = processHelmCharts(source, config?.registryAliases);
      if (dep) {
        dep.depType = depType;
        deps.push(dep);
      }
    }
```
Simplification with optional chaining.

File: lib/modules/manager/sveltos/extract.ts
Comment (secustor) [inline lib/modules/manager/sveltos/extract.ts:85]:
```suggestion
```

File: lib/modules/manager/sveltos/extract.ts
Comment (secustor) [inline lib/modules/manager/sveltos/extract.ts:62]: When would this be the case?

Comment (oliverbaehler) [reply lib/modules/manager/sveltos/extract.ts]: Never, leftover code

**Comment by oliverbaehler** [comment]
HI @secustor, I implemented your changes. I also moved the fixtures from the `__fixtures__` directory to inline consts since @rarkins  added a github check, that. fixtures no longer should be used. Trying to run:

```
Error: Cannot find module '@opentelemetry/semantic-conventions/incubating'
Require stack:
- /usr/src/app/lib/instrumentation/index.ts
- /usr/src/app/lib/renovate.ts
```

Lint also throws errors:

```
docker run -it --rm -v "${PWD}:/usr/src/app" -e LOG_LEVEL=debug -w /usr/src/app renovatebot_local  pnpm eslint . --cache --cache-location .cache/eslint --report-unused-disable-directives

> renovate@0.0.0-semantic-release eslint /usr/src/app
> eslint . --cache --cache-location .cache/eslint --report-unused-disable-directives "." "--cache" "--cache-location" ".cache/eslint" "--report-unused-disable-directives"


/usr/src/app/lib/expose.cjs
  1:1  error  Unused eslint-disable directive (no problems were reported from '@typescript-eslint/no-require-imports')

/usr/src/app/lib/instrumentation/index.ts
  25:3   error  ATTR_SERVICE_NAME not found in '@opentelemetry/semantic-conventions'               import/named
  26:3   error  ATTR_SERVICE_VERSION not found in '@opentelemetry/semantic-conventions'            import/named
  28:40  error  Unable to resolve path to module '@opentelemetry/semantic-conventions/incubating'  import/no-unresolved

/usr/src/app/lib/modules/manager/pipenv/artifacts.ts
  1:40  error  Unable to resolve path to module '@renovatebot/detect-tools'  import/no-unresolved

/usr/src/app/lib/modules/manager/pipenv/extract.ts
  1:40  error  Unable to resolve path to module '@renovatebot/detect-tools'  import/no-unresolved

/usr/src/app/lib/modules/platform/bitbucket-server/index.ts
  731:3  error  Unused eslint-disable directive (no problems were reported from '@typescript-eslint/only-throw-error')

/usr/src/app/lib/util/git/index.ts
  106:3  error  Unused eslint-disable directive (no problems were reported from '@typescript-eslint/only-throw-error')

/usr/src/app/lib/util/host-rules.ts
  113:17  error  Prefer using nullish coalescing operator (`??`) instead of a logical or (`||`), as it is a safer operator  @typescript-eslint/prefer-nullish-coalescing

/usr/src/app/lib/util/mutex.ts
  1:57  error  Unable to resolve path to module 'async-mutex'  import/no-unresolved

/usr/src/app/lib/util/result.spec.ts
  1:1  error  Unused eslint-disable directive (no problems were reported from '@typescript-eslint/prefer-promise-reject-errors')
  2:1  error  Unused eslint-disable directive (no problems were reported from '@typescript-eslint/only-throw-error')

/usr/src/app/lib/util/result.ts
  327:7  error  Unused eslint-disable directive (no problems were reported from '@typescript-eslint/only-throw-error')
  352:7  error  Unused eslint-disable directive (no problems were reported from '@typescript-eslint/only-throw-error')
  368:5  error  Unused eslint-disable directive (no problems were reported from '@typescript-eslint/only-throw-error')
  383:7  error  Unused eslint-disable directive (no problems were reported from '@typescript-eslint/only-throw-error')

/usr/src/app/lib/util/yaml.ts
   8:61  error  Unable to resolve path to module 'yaml'                    import/no-unresolved
  30:27  error  'any' overrides all other types in this intersection type  @typescript-eslint/no-redundant-type-constituents
  31:3   error  'any' overrides all other types in this intersection type  @typescript-eslint/no-redundant-type-constituents
  32:3   error  'any' overrides all other types in this intersection type  @typescript-eslint/no-redundant-type-constituents
  33:3   error  'any' overrides all other types in this intersection type  @typescript-eslint/no-redundant-type-constituents
  34:3   error  'any' overrides all other types in this intersection type  @typescript-eslint/no-redundant-type-constituents
```

DO I need to update/change something?

**Comment by secustor** [comment]
Looks like you forgot to install the updated dependencies, after merging in main.

**Comment by oliverbaehler** [comment]
@secustor Oh thanks! Sorry really not trying to waste everyone's time.. 3:

File: lib/modules/manager/sveltos/readme.md
Comment (HonkingGoose) [inline lib/modules/manager/sveltos/readme.md:45]:
```suggestion
| [EventTrigger](https://projectsveltos.github.io/sveltos/events/addon_event_deployment/#eventtrigger) | `EventTrigger`   |
```
Should we link to the section about `EventTrigger`s? Right now we're linking to the _general_ page for Events.

**Review by secustor — COMMENTED** [review]
needs prettier fix, else LGTM

**Review by secustor — APPROVED** [review]
Thanks for your contribution!

**Comment by oliverbaehler** [comment]
@secustor Thanks for merging, just a quick question: Does it take some time to update the version on developer.mend.io?

It keeps telling me that the configuration with the sveltos key is invalid 🤔:

```
INFO: Repository has invalid config
{
  "error": {
    "validationSource": "renovate.json",
    "validationError": "The renovate configuration file contains some invalid settings",
    "validationMessage": "Invalid configuration option: sveltos",
    "message": "config-validation",
    "stack": "Error: config-validation\n    at mergeRenovateConfig (/usr/local/renovate/lib/workers/repository/init/merge.ts:198:19)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at getRepoConfig (/usr/local/renovate/lib/workers/repository/init/config.ts:14:12)\n    at initRepo (/usr/local/renovate/lib/workers/repository/init/index.ts:56:12)\n    at Object.renovateRepository (/usr/local/renovate/lib/workers/repository/index.ts:64:14)\n    at attributes.repository (/usr/local/renovate/lib/workers/global/index.ts:202:11)\n    at start (/usr/local/renovate/lib/workers/global/index.ts:187:7)\n    at /usr/local/renovate/lib/renovate.ts:18:22"
  }
}
```

**Comment by secustor** [comment]
The hosted app is not updated automatically. Not sure when Mend will update the hosted app again.

File: lib/modules/manager/sveltos/extract.ts
Comment (viceice) [inline lib/modules/manager/sveltos/extract.ts:41]: wonder why we need this simple wrapper?

File: lib/modules/manager/sveltos/extract.ts
Comment (viceice) [inline lib/modules/manager/sveltos/extract.ts:54]: why trim trailing slash? doesn't make sense to me

File: lib/modules/manager/sveltos/extract.ts
Comment (viceice) [inline lib/modules/manager/sveltos/extract.ts:70]: this function should be inlined

File: lib/modules/manager/sveltos/util.ts
Comment (viceice) [inline lib/modules/manager/sveltos/util.ts:10]: same here, makes no sense and has no tests.

File: lib/modules/manager/sveltos/util.ts
Comment (viceice) [inline lib/modules/manager/sveltos/util.ts:8]:
```suggestion
  let modifiedChartName = chartName.substring(repoName.length). replace (regEx('^/'), '');
```
why not even simpler?

**Review by viceice — COMMENTED** [review]
sorry for late comments

**Pull request merged**
